File: 2377-check-if-number-has-equal-digit-count-and-digit-value/check-if-number-has-equal-digit-count-and-digit-value.py
class Solution:
    def digitCount(self, num):
        mp = {}
        for ch in num:
            mp[int(ch)] = mp.get(int(ch), 0) + 1
        for i in range(len(num)):
            if mp.get(i, 0) != int(num[i]):
                return False
        return True

# Example
print(Solution().digitCount("1210"))  # Output: True

File: 2377-check-if-number-has-equal-digit-count-and-digit-value/check-if-number-has-equal-digit-count-and-digit-value.cpp
class Solution {
public:
    bool digitCount(string num) {
        

        for(int i=0;i<num.size();i++){
            int s=count(num.begin(),num.end(),i+'0');
            if(num[i]-'0'!=s){
                return false;
            }

            
        }
        return true;

    }
};